aptos-move/block-executor/src/executor.rs

```diff
@@ -573,18 +578,13 @@ where
             // We return an error (leads to halting the execution) in the following cases:
             // 1) Code invariant violation.
             // 2) We detect module read/write intersection
-            // 3) A transaction triggered an Abort
-
-            last_input_output.record_finalized_group(txn_idx, finalized_groups);
-
+            // 3) A transaction triggered an Abort (unrecoverable VM error).
             // We cover cases 1 and 2 here
             maybe_code_error.and_then(|_| last_input_output.module_rw_intersection_ok())?;
-
             // Next, we handle 3, an abort / an unrecoverable VM error.
-            if let Some(err) = last_input_output.aborted_execution_status(txn_idx) {
-                assert!(matches!(err, BlockExecutionError::FatalVMError(_)));
-                return Err(err);
-            }
+            last_input_output.aborted_execution_status(txn_idx)?;
```

Comment:
this should now be name:
handle_aborted_execution_status or check_aborted_execution_status , as it doesn't return the status?

Reply:
Check makes sense, and if we fast-forward the abort, will fit even better.